src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs

```diff
@@ -134,25 +127,41 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
             throw ExceptionUtilities.Unreachable;
         }
 
-        internal static MethodSymbol? FindValidCloneMethod(NamedTypeSymbol containingType)
+        internal static MethodSymbol? FindValidCloneMethod(TypeSymbol containingType, ref HashSet<DiagnosticInfo>? useSiteDiagnostics)
         {
-            for (; !(containingType is null); containingType = containingType.BaseTypeNoUseSiteDiagnostics)
+            MethodSymbol? candidate = null;
+
+            foreach (var member in containingType.GetMembers(WellKnownMemberNames.CloneMethodName))
             {
-                foreach (var member in containingType.GetMembers(WellKnownMemberNames.CloneMethodName))
+                if (member is MethodSymbol
                 {
-                    if (member is MethodSymbol
-                    {
-                        DeclaredAccessibility: Accessibility.Public,
-                        IsStatic: false,
-                        ParameterCount: 0,
-                        Arity: 0
-                    } method && (method.IsOverride || method.IsVirtual || method.IsAbstract))
+                    DeclaredAccessibility: Accessibility.Public,
+                    IsStatic: false,
+                    ParameterCount: 0,
+                    Arity: 0
+                } method)
+                {
+                    if (candidate is object)
                     {
-                        return method;
+                        // An ammbiguity case, can come from metadata, treat as an error for simplicity.
+                        return null;
                     }
+
+                    candidate = method;
                 }
             }
-            return null;
+
+            if (candidate is null ||
+                !(containingType.IsSealed || candidate.IsOverride || candidate.IsVirtual || candidate.IsAbstract) ||
+                !containingType.IsEqualToOrDerivedFrom(
+                    candidate.ReturnType,
+                    TypeCompareKind.AllIgnoreOptions,
+                    ref useSiteDiagnostics))
+            {
```

Comment:
not blocking this PR: should we check that the return value isn't nullable? (updated the note in test plan with clone method) Also added a note for nullability in user-provided  `Deconstruct` method.

Reply:
> should we check that the return value isn't nullable? (updated the note in test plan with clone method) Also added a note for nullability in user-provided  Deconstruct method.

I do not believe nullable annotations should matter for the purpose of this method , they do not affect semantics.

---
In reply to: [456676833](https://github.com/dotnet/roslyn/pull/46053#discussion_r456676833) [](ancestors = 456676833)